src/libraries/System.Private.CoreLib/src/System/AppContext.AnyOS.cs

```diff
@@ -30,5 +31,34 @@ private static string GetBaseDirectoryCore()
 
             return directory;
         }
+
+        internal static void LogSwitchValues(RuntimeEventSource ev)
+        {
+            if (s_switches is not null)
+            {
+                lock (s_switches)
+                {
+                    foreach (var (k, v) in s_switches)
+                    {
+                        // Convert bool to int because it's cheaper to log (no boxing)
+                        ev.LogAppContextSwitch(k, v ? 0 : 1);
+                    }
+                }
+            }
+
+            if (s_dataStore is not null)
+            {
+                lock (s_dataStore)
+                {
+                    foreach (var (k, v) in s_dataStore)
+                    {
+                        if (v is string s && bool.TryParse(s, out bool isEnabled))
+                        {
+                            ev.LogAppContextSwitch(k, isEnabled ? 0 : 1);
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
The event doesn't seem to differentiate between switches that come from `s_dataStore` and `s_switches`, but I believe the lookup logic only falls back to `s_dataStore` if a requested key _doesn't_ exist in `s_switches`.

We should either differentiate between the two stores as part of the event or only send events for `s_dataStore` if a given key _doesn't_ exist in `s_swtiches`. Otherwise, we can see duplicate keys and have no way of knowing which took precedence.